scoring/score.py

```diff
@@ -1,20 +1,73 @@
+# -*- coding: utf-8 -*-
+
 import collections
 
 
 class InvalidScoresheetException(Exception):
     pass
 
 
+POINTS_HELD = 1
+POINTS_GROUND = 2
+POINTS_PLATFORM = 5
+
+NUM_TOKENS_IN_ARENA = 16
+
+
 class Scorer(object):
     def __init__(self, teams_data, arena_data):
         self._teams_data = teams_data
         self._arena_data = arena_data
 
+        for info in self._teams_data.values():
+            info.setdefault('tokens_held', 0)
+
+        for info in self._arena_data.values():
+            info.setdefault('tokens_ground', 0)
+            info.setdefault('tokens_platform', 0)
+
     def validate(self, extra):
-        pass
+        token_counts = (
+            [x['tokens_held'] for x in self._teams_data.values()] +
+            [extra.get('tokens_ground', 0)] +
+            [x['tokens_ground'] for x in self._arena_data.values()] +
+            [x['tokens_platform'] for x in self._arena_data.values()]
+        )
+
+        negative_counts = [x for x in token_counts if x < 0]
```

Comment:
`-0.0` doesn't get picked up by this, though I doubt it would make a difference in any reasonable code (especially since we're expecting integers)

Reply:
I think that's fine. I'm not sure what impact `-0.0` would have here; I'm reasonably sure it wouldn't impact anything, especially since we're now rejecting fractions too (c587c01).